client/src/components/rooms/RoomsMeetingExperience.tsx

```diff
@@ -43,17 +59,49 @@ export const RoomsMeetingExperience = (props: RoomsMeetingExperienceProps): JSX.
     _createAdapters();
   }, [credential]);
 
+  if (credential === undefined) {
+    return <>Failed to construct credential. Provided token is malformed.</>;
+  }
   if (callAdapter) {
     //TODO set forFactor to mobile
-    if (userRole === RoomParticipantRole.presenter) {
-      return <CallComposite adapter={callAdapter} callInvitationUrl={props.roomsInfo.inviteParticipantUrl} />;
-    } else {
-      return <CallComposite adapter={callAdapter} />;
-    }
-  }
 
-  if (credential === undefined) {
-    return <>Failed to construct credential. Provided token is malformed.</>;
+    switch (componentToShow(renderPostCall, userRole)) {
+      case PRESENTER:
+        return (
+          <CallComposite fluentTheme={theme} adapter={callAdapter} callInvitationUrl={roomsInfo.inviteParticipantUrl} />
+        );
+      case ATTENDEE:
+        return <CallComposite fluentTheme={theme} adapter={callAdapter} />;
+      case SURVEY:
+        if (postCall)
+          return (
+            <Survey
+              callId={callId}
+              acsUserId={userId}
+              meetingLink={locator.roomId}
+              theme={theme}
+              data-testid="Survey"
+              postCall={postCall}
+              onRejoinCall={async () => {
+                await callAdapter.joinCall();
+                setRenderPostCall(false);
+              }}
+            />
+          );
+        break;
+      default: {
+        // this is triggered in case postCall config is not specified. It will show default postCall screen - with rejoin button
+        if (userRole === RoomParticipantRole.presenter)
+          return (
```

Comment:
we can simplify this also to 
      <CallComposite
        adapter={callAdapter}
        fluentTheme={fluentTheme}
        formFactor={formFactorValue}
        callInvitationUrl={userRole === RoomParticipantRole.presenter ? roomsInfo.inviteParticipantUrl : undefined}
      />